GoogleForCreators/web-stories-wp
# Dashboard: Update dialogs to use new one from design system

## Summary
Updates dialogs that occur in dashboard to use new structure from design system. These are: Delete story dialog, preview story, delete publisher logo

## Relevant Technical Choices

- Swap existing dialog/modal and button/text elements used in dashboard for those in the design system
- Gut the dialog and modal components in dashboard
- Moved the titleFormatted function found in cardTitle and the story list to utils so i can use it in the delete story dialog without duplicating it (again). 
- Adds the background color used by story preview amp output as a theme value so that the backgrounds match in the dialog version of preview.


## To-do



## User-facing changes

- Dialogs in dashboard have been updated to implement new designs. 

## Testing Instructions

In the dashboard see the following: 

1. On "My stories": Click "Delete story" from the story menu in both grid and list view. See that dialog is visually updated (old on left).
![Screen Shot 2020-11-09 at 11 48 48 AM](https://user-images.githubusercontent.com/10720454/98583635-ea2ee700-2281-11eb-9867-ee0e65618fa7.png)

2. On "My stories": Click "Preview story" when hovering over a story (center action). See that dialog is visually updated (this one is a little unique since it's basically a glorified story preview that would open in a new tab as a modal instead. The biggest change here is the close icon is swapped for the new set. I didn't want to have to come back for it, made more sense to swap out at once). 
![Screen Shot 2020-11-09 at 11 50 18 AM](https://user-images.githubusercontent.com/10720454/98583662-f61aa900-2281-11eb-8370-b77ca55537c1.png)

3. On "Editor settings": click the pencil on a publisher logo (must have more than 1 logo uploaded) and then click "delete". See that the dialog is visually updated. 
![Screen Shot 2020-11-09 at 11 51 13 AM](https://user-images.githubusercontent.com/10720454/98583684-ff0b7a80-2281-11eb-9505-ae1a0a5413b1.png)

4. If a story has no title and you click the 'delete story' action, you'll get the matching "no title" title that displays in the grid or list view as the replacement title. 
![Screen Shot 2020-11-09 at 11 52 31 AM](https://user-images.githubusercontent.com/10720454/98583718-1185b400-2282-11eb-9011-684812c5cb39.png)

5. You can check story preview dialog views in storybook (dashboard/views/previewStory) to see loading and error states 
![preview states](https://user-images.githubusercontent.com/10720454/98583855-47c33380-2282-11eb-8f0c-a627657177a2.gif)

---



Addresses #4968
